taichi/backends/opengl/shaders/fast_pow.glsl.h

```diff
@@ -0,0 +1,38 @@
+// vim: ft=glsl
+// clang-format off
+#include "taichi/util/macros.h"
+STR(
+int fast_pow_i32(int x, int y)
+{
+  if (y > 512)
+    return int(pow(x, y));
+
+  bool neg = y < 0;
+  y = abs(y);
+  int ret = 1;
+  while (y != 0) {
+    if ((y & 1) != 0)
+      ret *= x;
+    x *= x;
+    y >>= 1;
+  }
+  return neg ? 1 / ret : ret;
```

Comment:
Does `1 / ret` make sense when `ret` is an integer?

Reply:
negative power for integer? It doesn't make sense anyway... For `x` is integer, let's assume `y >= 0` by `y = abs(y)`, wdyt?

Reply:
There doesn't seem to be a perfect way to solve this - maybe we should add a run-time assert in debug mode (`y >= 0` for `pow(int, int)`). I think we are good for now. Let's keep everything here.